src/Dependency/Model/AbstractClassDependency.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Symfony MakerBundle package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\MakerBundle\Dependency\Model;
+
+/**
+ * @author Jesse Rushlow <jr@rushlow.dev>
+ *
+ * @internal
+ */
+abstract class AbstractClassDependency
```

Comment:
Originally went with a single class that had `bool $isRequired` to determine if the dependency was required or optional. I went with an abstract base model and a `RequiredClassDependency` && `OptionalClassDependency` for better DX / code review readability... Atleast that was the intent..